test/e2e/lib/auditlogs.go

```diff
@@ -26,14 +26,15 @@ import (
 )
 
 func MakeAuditLogsOrDie(client *Client,
-	auditlogsName, methodName, project, resourceName, serviceName, targetName string,
+	auditlogsName, methodName, project, resourceName, serviceName, targetName, pubsubServiceAccount string,
```

Comment:
Not needed in this PR, just want to make sure people think about this (maybe make an issue)?

This is a lot of strings in a row. It will be hard/impossible for someone reading the code to see that everything is in the correct position. I recommend creating a struct instead of passing seven strings in a row.

A similar problem exists on most of the `MakeFoo(...)` style methods in this change.

Reply:
The MakeFoo(…) style method looks like:

https://github.com/google/knative-gcp/blob/01e5d59d820e7918fe4f5d557dc669bdfac45eea/test/e2e/lib/auditlogs.go#L28-L42

Those strings inside the MakeFoo(…) will be transformed to source options, and it accepts other options. 
I feel I should not add a new input variable, but using format like following instead.
`lib.MakeAuditLogsOrDie(client,
   auditlogsName, 
   methodName, 
   project, 
   resourceName, 
   serviceName, 
   targetName,
   kngcptesting.WithCloudStorageSourceGCPServiceAccount(pubsubServiceAccount)))`
(I was wondering if the existing input strings are the basic/required part for making a source). I am fine to make them as a struct. But should I include the pubsubServiceAccount in the struct? Or use it just like `kngcptesting.WithCloudStorageSourceGCPServiceAccount(pubsubServiceAccount)`?

Reply:
I opened https://github.com/google/knative-gcp/issues/764 for this

Reply:
My intention was something more like:

```go
struct AuditLogsConfig {
  // Possibly include Client too.
  auditlogsName string
  methodName string
  project string
  resourceName string
  serviceName string
  targetName string
  pubsubServiceAccount string
  options []kngcptesting.CloudAuditLogsSourceOption
}

func MakeAuditLogsOrDie(client *Client, cfg AuditLogsConfig) { ... }

// Then calling it would be something like:
MakeAuditLogsOrDie(client, AuditLogsConfig {
  auditlogsName: auditlogsName,
  methodName: methodName,
  project: project,
  resourceName: resourceName,
  serviceName: serviceName,
  targetName: targetName,
  pubsubServiceAccount: pubsubServiceAccount,
  options: options,
})
```

That way the strings are named, so it is easier to see if something is out of order.

Reply:
Yeah, so I was wondering if I need to do like this? (not including `pubsubServiceAccount` as a part of that struct, as I was wondering if the rest input strings are the basic/required part for making a source, and the `pubsubServiceAccount` input I added in this PR is not required.)
```
MakeAuditLogsOrDie(client, AuditLogsConfig {
  auditlogsName: auditlogsName,
  methodName: methodName,
  project: project,
  resourceName: resourceName,
  serviceName: serviceName,
  targetName: targetName,
  options: []kngcptesting.CloudAuditLogsSourceOption{kngcptesting.WithCloudStorageSourceGCPServiceAccount(pubsubServiceAccount),}
})```